File: 1768.go
func mergeAlternately(word1 string, word2 string) string {
  len1 := len(word1)
  len2 := len(word2)
  var result string
  for i := 0; i < len1 || i < len2; i++ {
    if i < len1 {
      result += string(word1[i])
    } else {
      result += string(word2[i:])
      break
    }
    if i < len2 {
      result += string(word2[i])
    } else {
      result += string(word1[i+1:])
      break
    }
  }
  return result
}

File: 1768.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
      auto size1 = word1.size();
      auto size2 = word2.size();
      auto count = std::max(size1, size2);
      string result;
      for (auto i = 0; i < count; i++) {
        if (i < size1)
          result += word1[i];
        if (i < size2)
          result += word2[i];
      }
      return result;
    }
};